chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/support/ProcessPeaksByAreaFilterUtils.java

```diff
@@ -14,43 +14,96 @@
 import java.util.Collection;
 
 import org.eclipse.chemclipse.model.core.IPeak;
-import org.eclipse.chemclipse.xxd.model.filter.peaks.ProcessPeaksByAreaFilterSettings;
+import org.eclipse.chemclipse.model.core.IPeakModel;
+import org.eclipse.chemclipse.processing.filter.CRUDListener;
+import org.eclipse.chemclipse.xxd.model.filter.peaks.ProcessPeaksByIntegratedAreaFilterSettings;
+import org.eclipse.chemclipse.xxd.model.filter.peaks.ProcessPeaksByPercentageAreaFilterSettings;
 
 public class ProcessPeaksByAreaFilterUtils {
 
-	public static double[] getAreaLimits(ProcessPeaksByAreaFilterSettings configuration, ProcessPeaksByAreaFilterAreaTypeSelection selection) {
-		
-		double[] areaLimits = new double[2];
-		if(selection==ProcessPeaksByAreaFilterAreaTypeSelection.INTEGRATED_AREA) {
-			areaLimits[0] = configuration.getMinimumAreaValue();
-			areaLimits[1] = configuration.getMaximumAreaValue();
-		} else {
-			areaLimits[0] = configuration.getMinimumPercentageAreaValue();
-			areaLimits[1] = configuration.getMaximumPercentageAreaValue();
-		}
-		if(areaLimits[0]>areaLimits[1]) {
-			throw new IllegalArgumentException("Please check the values of the specified limits!");
+	public static double[] getIntegratedAreaLimits(ProcessPeaksByIntegratedAreaFilterSettings configuration) {
+
+		if(configuration.getMinimumAreaValue()>configuration.getMaximumAreaValue()) {
+			throw new IllegalArgumentException("Please check the values of the integrated area limits!");
 		}
-		return areaLimits;
+		return new double[] {configuration.getMinimumAreaValue(), configuration.getMaximumAreaValue()};
 	}
 
-	public static <X extends IPeak> double calculateAreaCompareValue(ProcessPeaksByAreaFilterAreaTypeSelection selection, X peak, Collection<X> read) {
-		
-		double compareValue;
-		if(selection==ProcessPeaksByAreaFilterAreaTypeSelection.INTEGRATED_AREA) {
-			compareValue = peak.getIntegratedArea();
-		} else {
-			compareValue = (100/calculateAreaSum(read))*peak.getIntegratedArea();
+	public static double[] getPercentageAreaLimits(ProcessPeaksByPercentageAreaFilterSettings configuration) {
+
+		if(configuration.getMinimumPercentageAreaValue()>configuration.getMaximumPercentageAreaValue()) {
+			throw new IllegalArgumentException("Please check the values of the percentage area limits!");
 		}
-		return compareValue;
+		return new double[] {configuration.getMinimumPercentageAreaValue(), configuration.getMaximumPercentageAreaValue()};
+	}
+
+	public static <X extends IPeak> double calculateIntegratedAreaCompareValue(X peak, Collection<X> read) {
+
+		return peak.getIntegratedArea();
+	}
+
+	public static <X extends IPeak> double calculatePercentageAreaCompareValue(X peak, Collection<X> read) {
+
+		return (100 / calculateAreaSum(read)) * peak.getIntegratedArea();
 	}
 
 	private static <X extends IPeak> double calculateAreaSum(Collection<X> read) {
-		
-		double areaSum = 0;		
+
+		double areaSum = 0;
```

Comment:
For every peak, the area sum is here calculated over and over again.
I would rather calculate that sum once and re-use.
for the abs area filter, you dont need that code at all. try to get rid of the utils class and simplify the code a bit.